lib/rucio/daemons/storage/consistency/actions.py

```diff
@@ -0,0 +1,805 @@
+          force_proceed, scanner_files_path)
+
+def actions_loop(once, rses, sleep_time, dark_min_age, dark_threshold_percent,
+  miss_threshold_percent, force_proceed, scanner_files_path):
+
+    """
+    Main loop to apply the CC actions
+    """
+
+    hostname = socket.gethostname()
+    pid = os.getpid()
+    current_thread = threading.current_thread()
+
+    executable = 'storage-consistency-actions'
+    heartbeat = live(executable=executable, hostname=hostname, pid=pid, thread=current_thread)
+
+    # Make an initial heartbeat
+    # so that all storage-consistency-actions have the correct worker number on the next try
+    prefix = 'storage-consistency-actions[%i/%i] ' % (heartbeat['assign_thread'],
+      heartbeat['nr_threads'])
+    logger = formatted_logger(logging.log, prefix + '%s')
+    logger(logging.INFO, '\n hostname: %s  pid: %d  current_thread: %s' %
+     (hostname, pid, current_thread))
+
+    graceful_stop.wait(1)
+
+    while not graceful_stop.is_set():
+        try:
+            heartbeat = live(executable=executable, hostname=hostname, pid=pid,
+              thread=current_thread)
+            logger(logging.INFO, '\n heartbeat? %s' % heartbeat)
+            
+            prefix = 'storage-consistency-actions[%i/%i] ' % (heartbeat['assign_thread'],
+              heartbeat['nr_threads'])
+            logger(logging.INFO, '\n prefix: %s' % prefix)
+            start = time.time()
+            logger(logging.DEBUG, 'Start time: %f' % start)
+
+            deckard_loop(rses, dark_min_age, dark_threshold_percent, miss_threshold_percent,
+              force_proceed, scanner_files_path)
+            daemon_sleep(start_time=start, sleep_time=sleep_time, graceful_stop=graceful_stop,
+              logger=logger)
+
+
+        except Exception as e:
+            traceback.print_exc()
+            logger(logging.WARNING, '\n Something went wrong here... %s' % e)
+            logger(logging.WARNING, '\n Something went wrong here... %s ' % (e.__class__.__name__))
+        if once:
+            break
+
+    die(executable=executable, hostname=hostname, pid=pid, thread=current_thread)
+
+
+def stop(signum=None, frame=None):
+    """
+    Graceful exit.
+    """
+    graceful_stop.set()
+
+
+def run(once=False, rses=None, sleep_time=60, default_dark_min_age=28, default_dark_threshold_percent=1.0,
+  default_miss_threshold_percent=1.0, force_proceed=False, default_scanner_files_path="/var/cache/consistency-dump",
+  threads=1):
+    """
+    Starts up the Consistency-Actions.
+    """
+
+# Setup logging
+    setup_logging()
+
+    prefix = 'storage-consistency-actions (run())'
+    logger = formatted_logger(logging.log, prefix + '%s')
+
+    logger(logging.INFO, '\n Now inside run(once...)\n')
+
+    dark_min_age = default_dark_min_age
+    dark_threshold_percent = default_dark_threshold_percent
+    miss_threshold_percent = default_miss_threshold_percent
+    scanner_files_path = default_scanner_files_path
```

Comment:
Redundant, there is no reason to create new variables.
Consider replacing the `default_` variables with their significant other

Reply:
I've added an explanation as to why this is done. 
And used a # TODO codetag, so that this gets fixed in the future.